src/Features/Core/Portable/Shared/Utilities/PatternMatcher.cs

```diff
@@ -256,11 +257,12 @@ public IEnumerable<PatternMatch> GetMatches(string candidate, string dottedConta
 
             // So far so good.  Now break up the container for the candidate and check if all
             // the dotted parts match up correctly.
-            var totalMatch = candidateMatch.ToList();
+            var totalMatch = new List<PatternMatch>();
 
```

Comment:
I realize it's not a new allocation, but should this be pooled in the pattern matcher?